mempool/mempool.go

```diff
@@ -393,33 +393,34 @@ func (mem *Mempool) Reap(maxTxs, maxTxsBytes int) types.Txs {
 		time.Sleep(time.Millisecond * 10)
 	}
 
-	txs := mem.collectTxs(maxTxs, maxTxsBytes)
+	var cur int
+	txs := make([]types.Tx, 0, mem.txs.Len())
```

Comment:
I think we will get a performance boost if we have a good estimate of avg size per tx, and set the initial capacity based off of that. (We could under-esitmate avg size per tx a bit) In the case of max=-1, we could just call reapTxs(-1) for simplicity.

Reply:
Indeed, but how big of a boost? It's better to make such optimizations after we know current performance profile. It may be that the time to calculate avg size will hurt us more (overweights gains).

Reply:
Great point!